xarray/core/computation.py

```diff
@@ -789,9 +789,9 @@ def apply_ufunc(
         the style of NumPy universal functions [1]_ (if this is not the case,
         set ``vectorize=True``). If this function returns multiple outputs, you
         must set ``output_core_dims`` as well.
-    *args : Dataset, DataArray, GroupBy, Variable, numpy/dask arrays or scalars
+    *args : Dataset, DataArray, GroupBy, Variable, numpy.ndarray, dask.array.Array or scalar
         Mix of labeled and/or unlabeled arrays to which to apply the function.
-    input_core_dims : Sequence[Sequence], optional
+    input_core_dims : sequence of sequence, optional
```

Comment:
Do we have any preference on whether to use numpydoc type specs or type hints?

If we choose to use type hints (exclusively or in combination with numpydoc type specs), we might have to wait a bit: `napoleon`'s type preprocessor currently doesn't work well with them, but support for them will probably be included in sphinx 3.3.0 (scheduled for October).

Reply:
I think this is fine.